app/controllers/api/articles_controller.rb

```diff
@@ -17,9 +19,21 @@ def index
 
   def show
     article = Article.find(params[:id])
-
     render json: article, serializer: ArticlesShowSerializer
   rescue ActiveRecord::RecordNotFound => e
     render json: { error_message: "Article does not exist" }, status: 404
   end
+
+  def create
+    unless current_user.role == "journalist"
```

Comment:
If you are using `enum` you can simplify this

```ruby
current_user.journalist?
```

Reply:
Good point!